File: services/cpp-worker/plugins/hello_plugin.cpp
#include "../src/plugin.h"
#include <chrono>
#include <iomanip>
#include <sstream>

class HelloCppPlugin : public Plugin {
public:
    std::string get_name() const override {
        return "hello_cpp";
    }

    std::string get_description() const override {
        return "Returns a hello message from C++ worker";
    }

    json execute(const json& params, ExecutionContext* context) override {
        std::string name = "World";
        if (params.contains("name") && params["name"].is_string()) {
            name = params["name"].get<std::string>();
        }

        // Get current timestamp
        auto now = std::chrono::system_clock::now();
        auto time_t_now = std::chrono::system_clock::to_time_t(now);
        std::stringstream ss;
        ss << std::put_time(std::gmtime(&time_t_now), "%Y-%m-%dT%H:%M:%SZ");

        return {
            {"message", "Hello " + name + " from C++! 🔷"},
            {"worker_id", context->worker_id},
            {"timestamp", ss.str()},
            {"cpp_version", __cplusplus},
            {"status", "success"}
        };
    }
};

// Factory function
PluginPtr create_hello_cpp_plugin() {
    return std::make_shared<HelloCppPlugin>();
}

File: services/node-worker/plugins/hello-plugin.js
const BasePlugin = require('./base-plugin');

/**
 * Hello Plugin - Simple greeting capability
 */
class HelloPlugin extends BasePlugin {
    getName() {
        return "hello_node";
    }

    getDescription() {
        return "Returns a hello message from Node.js worker";
    }

    getHttpMethod() {
        return "POST";
    }

    async execute(params, context) {
        const name = params.name || "World";
        
        return {
            message: `Hello ${name} from Node.js! 🟢`,
            worker_id: context.workerId,
            timestamp: new Date().toISOString(),
            node_version: process.version,
            status: "success"
        };
    }
}

module.exports = HelloPlugin;
